src/components/shortcircuit-analysis-result.js

```diff
@@ -4,17 +4,93 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-import React from 'react';
+import React, { useCallback, useMemo, useRef } from 'react';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
 import { useSelector } from 'react-redux';
-import VirtualizedTable from './util/virtualized-table';
+import { AgGridReact } from 'ag-grid-react';
+import { makeStyles, useTheme } from '@mui/styles';
+import 'ag-grid-community/styles/ag-grid.css';
+import 'ag-grid-community/styles/ag-theme-alpine.css';
+import clsx from 'clsx';
+import { groupPostSort } from './util/sort-functions';
+
+const useStyles = makeStyles((theme) => ({
+    grid: {
+        width: 'auto',
+        height: '100%',
+        position: 'relative',
+    },
+}));
 
 const ShortCircuitAnalysisResult = ({ result }) => {
     const intl = useIntl();
+    const gridRef = useRef();
 
     const shortCircuitNotif = useSelector((state) => state.shortCircuitNotif);
 
+    const theme = useTheme();
+    const GRID_PREFIX = 'grid.';
+    const classes = useStyles();
+
+    const columns = useMemo(() => {
+        return [
+            {
+                headerName: intl.formatMessage({ id: 'IDNode' }),
+                field: 'elementId',
+            },
+            {
+                headerName: intl.formatMessage({ id: 'Type' }),
+                field: 'faultType',
+            },
+            {
+                headerName: intl.formatMessage({ id: 'Feeders' }),
+                field: 'connectableId',
+            },
+            {
+                headerName: intl.formatMessage({ id: 'IscKA' }),
+                field: 'current',
+            },
+
+            {
+                headerName: intl.formatMessage({ id: 'LimitType' }),
+                field: 'limitType',
+            },
+
+            {
+                headerName: intl.formatMessage({ id: 'IscMinKA' }),
+                field: 'limitMin',
+            },
+            {
+                headerName: intl.formatMessage({ id: 'IscMaxKA' }),
+                field: 'limitMax',
+            },
+            {
+                headerName: intl.formatMessage({ id: 'PscMVA' }),
+                field: 'shortCircuitPower',
+            },
+        ];
+    }, [intl]);
+
+    const getRowStyle = useCallback(
+        (params) => {
+            if (params?.data?.elementId) {
+                return {
+                    backgroundColor: theme.selectedRow.background,
+                };
+            }
+        },
+        [theme.selectedRow.background]
+    );
+
+    const getLocaleText = useCallback(
```

Comment:
to refactor with 
src/components/spreadsheet/equipment-table.js:    const getLocaleText = useCallback(